src/Util/Logger/WorkerLogger.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+namespace Friendica\Util\Logger;
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * A Logger for specific Worker, which adds an additional id to it.
+ *
+ * uses the Decorator pattern (https://en.wikipedia.org/wiki/Decorator_pattern)
+ */
+class WorkerLogger implements LoggerInterface
+{
+	/**
+	 * @var LoggerInterface The original logger
+	 */
+	private $logger;
+
+	/**
+	 * @var string The worker ID
+	 */
+	private $workerId;
+
+	/**
+	 * @var string The current function Name of the worker
+	 */
+	private $functionName;
+
+	public function __construct(LoggerInterface $logger, $functionName)
+	{
+		$this->logger = $logger;
+		$this->functionName = $functionName;
+		$this->workerId = $this->generateUid(7);
+	}
+
+	/**
+	 * Generates an UID
+	 *
+	 * @param $length
+	 * @return bool|string
+	 */
+	private function generateUid($length)
+	{
+		return substr(hash('md5', uniqid('', true)), 0, $length);
```

Comment:
If you aren't using the output of `uniqid()` as is, so you don't need it.

Please use something simpler like `bin2hex(random_bytes($length / 2))` instead of `hash('md5', uniqid('', true))`.